app/src/main/java/com/google/droidjump/leveldata/LevelData.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.droidjump.leveldata;
+
+import android.content.res.Resources;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.LinkedList;
+import java.util.Scanner;
+
+public class LevelData {
+    final static String baseSpeedKey = "baseSpeed";
+    final static String timelineKey = "timeline";
+    final static String intervalKey = "interval";
+    final static String typeKey = "type";
+
+    private int baseSpeed;
+    private LinkedList<ObstacleData> obstaclesData;
+
+    public LevelData(Level level, Resources resources) {
+        obstaclesData = new LinkedList<>();
+        getDataFromFile(level, resources);
+    }
+
+    private void getDataFromFile(Level level, Resources resources) {
+        try {
+            JSONObject leveldata = new JSONObject(getJSONStringFromResource(level.fileId, resources));
+            baseSpeed = leveldata.getInt(baseSpeedKey);
+            JSONArray timeline = leveldata.getJSONArray(timelineKey);
+            for (int i = 0; i < timeline.length(); i++) {
+                JSONObject currentObject = timeline.getJSONObject(i);
+                int interval = currentObject.getInt(intervalKey);
+                ObstacleType type = Enum.valueOf(ObstacleType.class, currentObject.getString(typeKey));
+                obstaclesData.add(new ObstacleData(interval, type));
+            }
+        } catch (JSONException e) {
+            e.printStackTrace();
```

Comment:
When a exception is thrown is better to log here somehting instead of just ignoring it. I would add a log statement using Log class. Log.d()/Log.e(TAG, error message). Usually the TAG is the name of the class.

Reply:
Added, is it correct? and should I remove e.printStackTrace() ?

Reply:
Done